action/protocol/staking/vote_reviser.go

```diff
@@ -96,10 +96,17 @@ func (vr *VoteReviser) calculateVoteWeight(csm CandidateStateManager) (Candidate
 		}
 
 		if cand.SelfStakeBucketIdx == bucket.Index {
-			cand.AddVote(calculateVoteWeight(vr.c, bucket, true))
+			err = cand.AddVote(calculateVoteWeight(vr.c, bucket, true))
+			if fixUnproductiveDelegates && err != nil {
```

Comment:
vote reviser are only used on `Greenland` and `Hawaii` heights, which are already past heights, so i think you can directly attempt to fix here:
```
if err := cand.AddVote(calculateVoteWeight(vr.c, bucket, true)); err != nil {
}
```
and run a fullsync test of your PR, it is expected to pass the fullsync

Reply:
I see,  👍